hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-nativetask/src/test/java/org/apache/hadoop/mapred/nativetask/kvtest/KVTest.java

```diff
@@ -89,31 +87,33 @@ public static Iterable<Class<?>[]> data() throws Exception {
     // Generate an entry for each key type.
     List<Class<?>[]> pairs = Lists.newArrayList();
     for (Class<?> keyClass : keyClasses) {
-      pairs.add(new Class<?>[]{ keyClass, LongWritable.class });
+      pairs.add(new Class<?>[]{keyClass, LongWritable.class});
     }
     // ...and for each value type.
     for (Class<?> valueClass : valueClasses) {
-      pairs.add(new Class<?>[]{ LongWritable.class, valueClass });
+      pairs.add(new Class<?>[]{LongWritable.class, valueClass});
     }
     return pairs;
   }
 
-  private final Class<?> keyclass;
-  private final Class<?> valueclass;
+  private Class<?> keyclass;
+  private Class<?> valueclass;
 
-  public KVTest(Class<?> keyclass, Class<?> valueclass) {
+  public void initKVTest(Class<?> keyclass, Class<?> valueclass) {
     this.keyclass = keyclass;
     this.valueclass = valueclass;
   }
 
-  @Before
+  @BeforeEach
   public void startUp() throws Exception {
-    Assume.assumeTrue(NativeCodeLoader.isNativeCodeLoaded());
-    Assume.assumeTrue(NativeRuntime.isNativeLibraryLoaded());
+    Assumptions.assumeTrue(NativeCodeLoader.isNativeCodeLoaded());
+    Assumptions.assumeTrue(NativeRuntime.isNativeLibraryLoaded());
   }
 
-  @Test
-  public void testKVCompability() throws Exception {
+  @MethodSource("data")
+  @ParameterizedTest(name = "key:{0}\nvalue:{1}")
```

Comment:
Could you verify this name setting works as expected? https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5021/1/testReport/org.apache.hadoop.mapred.nativetask.kvtest/KVTest/

Before this change, the classes are displayed https://ci-hadoop.apache.org/job/hadoop-qbt-trunk-java8-linux-x86_64/1106/testReport/org.apache.hadoop.mapred.nativetask.kvtest/KVTest/